src/Microsoft.Data/DataFrame.cs

```diff
@@ -33,6 +36,273 @@ public DataFrame(IList<BaseColumn> columns)
+            }
+            Schema schema = schemaBuilder.Build();
+            List<Apache.Arrow.Array> arrays = new List<Apache.Arrow.Array>();
+
+            for (int n = 0; n < numberOfRecordBatches; n++)
+            {
+                for (int i = 0; i < numberOfColumns; i++)
+                {
+                    BaseColumn column = Column(i);
+                    //Memory<byte> values, nulls;
+                    //ArrowBuffer valueBuffer, nullBuffer;
+                    switch (column)
+                    {
+                        case PrimitiveColumn<bool> boolColumn:
+                            arrays.Add(boolColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<byte> byteColumn:
+                            arrays.Add(byteColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<char> charColumn:
+                            arrays.Add(charColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<decimal> decimalColumn:
+                            arrays.Add(decimalColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<double> doubleColumn:
+                            arrays.Add(doubleColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<float> floatColumn:
+                            arrays.Add(floatColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<int> intColumn:
+                            arrays.Add(intColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<long> longColumn:
+                            arrays.Add(longColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<sbyte> sbyteColumn:
+                            arrays.Add(sbyteColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<short> shortColumn:
+                            arrays.Add(shortColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<uint> uintColumn:
+                            arrays.Add(uintColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<ulong> ulongColumn:
+                            arrays.Add(ulongColumn.AsArrowArray(n));
+                            break;
+                        case PrimitiveColumn<ushort> ushortColumn:
+                            arrays.Add(ushortColumn.AsArrowArray(n));
+                            break;
+                        default:
+                            throw new NotImplementedException(nameof(column.DataType));
+                    }
+                }
+                yield return new RecordBatch(schema, arrays, 0);
```

Comment:
Ignore the 0 here. I didn't spend much time here thinking about how to split up each `PrimitiveColumn<T>`'s `DataFrameBuffer`s such that there is a 1x1 match with their `NullBitMapBuffer`s. For this test, each column in the DataFrame becomes an Arrow `Array`